### ISSUE TITLE
Unexpected behavior in post-processing when using custom response distribution

### ISSUE THREAD

**Issue by RaoulWolf** [paul-buerkner/brms#909]
_**tl;dr**
post-processing behaves differently (or fails) when using custom response distribution._

This comes on the heels of #906, and might be more related to the way brms interacts with bayesplot and loo (and potentially other packages), so I'm opening a separate issue. 

Using a custom response distribution in a multivariate model now works like charm using brms version 2.12.11 (same example as in #906):
```
phi <- 0.7
n <- 300
z <- rnorm(n, sd = 0.2)
ntrials <- sample(1:10, n, replace = TRUE)
eta <- 1 + z
mu <- exp(eta) / (1 + exp(eta))
a <- mu * phi
b <- (1 - mu) * phi
p <- rbeta(n, a, b)
y <- rbinom(n, ntrials, p)
l <- rlnorm(n)
cat <- sample(c("a", "b"), n, replace = TRUE)

dat <- data.frame(y, z, ntrials, l, cat)

## Definition of custom response distribution
beta_binomial2 <- custom_family(
  "beta_binomial2", dpars = c("mu", "phi"),
  links = c("logit", "log"), lb = c(NA, 0),
  type = "int", vars = "vint1[n]"
)

stan_funs <- "
  real beta_binomial2_lpmf(int y, real mu, real phi, int T) {
    return beta_binomial_lpmf(y | T, mu * phi, (1 - mu) * phi);
  }
  int beta_binomial2_rng(real mu, real phi, int T) {
    return beta_binomial_rng(T, mu * phi, (1 - mu) * phi);
  }
"

stanvars <- stanvar(scode = stan_funs, block = "functions")

## Model fit
fit <- brm(mvbf(bf(formula = y | vint(ntrials) ~ z, flist = list(phi ~ cat), family = beta_binomial2),
                bf(formula = l ~ z, flist = list(sigma ~ cat), family = lognormal)),
           data = dat, stanvars = stanvars)
```

The model is parsed successfully and even converges without any apparent issues. 

The post-processing (as showcased in the ["Define Custom Response Distributions with brms" vignette](https://cran.r-project.org/web/packages/brms/vignettes/brms_customfamilies.html) now works somehow in version 2.12.11; however, the behavior of the posterior checks is different than expected. After running the necessary code for post-processing:
```
expose_functions(fit, vectorize = TRUE)

log_lik_beta_binomial2 <- function(i, prep) {
  mu <- prep$dpars$mu[, i]
  phi <- prep$dpars$phi
  trials <- prep$data$vint1[i]
  y <- prep$data$Y[i]
  beta_binomial2_lpmf(y, mu, phi, trials)
}

posterior_predict_beta_binomial2 <- function(i, prep, ...) {
  mu <- prep$dpars$mu[, i]
  phi <- prep$dpars$phi
  trials <- prep$data$vint1[i]
  beta_binomial2_rng(mu, phi, trials)
}

posterior_epred_beta_binomial2 <- function(prep) {
  mu <- prep$dpars$mu
  trials <- prep$data$vint1
  trials <- matrix(trials, nrow = nrow(mu), ncol = ncol(mu), byrow = TRUE)
  mu * trials
}
```

I noticed the following: 

* `loo(fit)` takes a very (!) long time to run and eventually crashes RStudio on my machine.
* `pp_check(fit, resp = "y")` takes longer, plots _all_ draws and seemingly ignores the `nsamples` parameter, while `pp_check(fit, resp = "l")` behaves as expected.
* `conditional_effects(fit)` fails with the following error message:
```
Error: 'vint' requires whole numbers as input.
```

This also occurs when using a custom response distribution in a normal uni-variate model, so it's not limited to multivariate models. 

p.s.: bayesplot version 1.7.1 and loo version 2.2.0, both from CRAN.

**Comment by paul-buerkner** [comment]
Thank you for opening this issue. We have two separate problems here. The first one, with the slow computation is because the post-processing functions I presented in the vignette are not desigend for `phi` to be predicted as well. To make this work, use `phi <- prep$dpars$phi[, i]` or branch appropriate with an if statement to support both predicted and non-predicted `phi` at the same time. I have added a footnote to the vignette.

The second issue is with the handling of `vint` in `conditional_effects`, which I have fixed.
